src/rules/prefer-importing-jest-globals.ts

```diff
@@ -0,0 +1,189 @@
+          return;
+        }
+        const jestFunctionsToImport = jestFunctionsToReport.map(
+          jestFunction => jestFunction.name,
+        );
+        const reportingNode = jestFunctionsToReport[0].head.node;
+
+        const jestFunctionsToImportFormatted = jestFunctionsToImport.join(', ');
+
+        const isModule = context.parserOptions.sourceType === 'module';
+
+        context.report({
+          node: reportingNode,
+          messageId: 'preferImportingJestGlobal',
+          data: { jestFunctions: jestFunctionsToImportFormatted },
+          fix(fixer) {
+            const sourceCode = getSourceCode(context);
+            const [firstNode] = sourceCode.ast.body;
+
+            // check if "use strict" directive exists
+            if (
+              firstNode.type === AST_NODE_TYPES.ExpressionStatement &&
+              firstNode.expression.type === AST_NODE_TYPES.Literal &&
+              firstNode.expression.value === 'use strict'
+            ) {
+              return fixer.insertTextAfter(
+                firstNode,
+                `\n${createFixerImports(isModule, jestFunctionsToImport)}`,
+              );
+            }
+
+            const importNode = sourceCode.ast.body.find(
+              node =>
+                node.type === AST_NODE_TYPES.ImportDeclaration &&
+                node.source.value === '@jest/globals',
+            );
+
+            if (
+              importNode &&
+              importNode.type === AST_NODE_TYPES.ImportDeclaration
+            ) {
+              const existingImports = importNode.specifiers.reduce<string[]>(
+                (imports, specifier) => {
+                  /* istanbul ignore else */
+                  if (
+                    specifier.type === AST_NODE_TYPES.ImportSpecifier &&
+                    specifier.imported?.name
+                  ) {
+                    imports.push(specifier.imported.name);
+                  }
+
+                  return imports;
+                },
+                [] as string[],
+              );
+
+              const allImports = [
+                ...new Set([...existingImports, ...jestFunctionsToImport]),
+              ];
+
+              return fixer.replaceText(
+                importNode,
+                createFixerImports(isModule, allImports),
+              );
+            }
+
+            const requireNode = sourceCode.ast.body.find(
+              node =>
+                node.type === AST_NODE_TYPES.VariableDeclaration &&
+                node.declarations.some(
+                  declaration =>
+                    declaration.init &&
+                    (declaration.init as any).callee &&
+                    (declaration.init as any).callee.name === 'require' &&
+                    (declaration.init as any).arguments?.[0]?.type ===
+                      'Literal' &&
+                    (declaration.init as any).arguments?.[0]?.value ===
+                      '@jest/globals',
```

Comment:
Don't cast to `any` - even if you have to `istanbul ignore`, properly narrow these down through checks; this is a `CallExpression`, which you want to know if it has a single argument and if that argument is a literal with the value `@jest/globals`.

We have utilities to help you with this - checkout `hasOnlyOneArgument` and `isStringNode`; that should also lead you to realise there are holes you need to explore, like what if it's a template literal (which is a different type but covered by using `isStringNode`), template literal with interpolation, a variable, and so on.

Reply:
Cool, I could remove the cast to `any`. But I have type issue with the helper function. Could you come up with a code suggestion please?